### PR TITLE
Fix for issue #66, removing 'nose' as a production requirement.

### PR DESCRIPTION
**Pull request by dwvisser**
This fixes the issue #66 where `nose` is getting installed automatically when someone invokes `pip install inotify`. Instead, it places `nose` in the test dependencies and `setuptools` test runner target.

### CODE REVIEW

**Comment by skonakov-tophat** [comment]
Running into the same issue, would be awesome to get this pr merged :)

**Comment by dsoprea** [comment]
I'll merge it if you can provide a URL that explains how and where those testing-dependencies are installed if not in the general requirements.

**Comment by nehaljwani** [comment]
>I'll merge it if you can provide a URL that explains how and where those testing-dependencies are installed if not in the general requirements.

From https://setuptools.readthedocs.io/en/stable/setuptools.html:
>tests_require
If your project’s tests need one or more additional packages besides those needed to install it, you can use this option to specify them. It should be a string or list of strings specifying what other distributions need to be present for the package’s tests to run. When you run the test command, setuptools will attempt to obtain these (even going so far as to download them using EasyInstall). Note that these required projects will not be installed on the system where the tests are run, but only downloaded to the project’s setup directory if they’re not already installed locally.

So, basically if you run: `python setup.py test`, the test requirements will be downloaded.
